core/store/src/trie/shard_tries.rs

```diff
@@ -181,45 +181,18 @@ impl ShardTries {
         &self.0.store.storage
     }
 
-    pub(crate) fn update_cache(&self, transaction: &DBTransaction) -> std::io::Result<()> {
+    pub fn update_cache(&self, changes: &[TrieRefcountChange], shard_uid: ShardUId) {
```

Comment:
Quick comment: it looks like that we always insert items to cache even if there was a deletion. Maybe conversion from `TrieRefcountChange` to `ops` should happen on `apply_insertions/deletions` level.

Maybe it is worth including some test here, involving:
* creating `ShardTries`
* updating cache inserting and deleting some trie changes
* checking that inserted values are present and deleted are not.
But it's not trivial, because caches are impl detail of `ShardTries`. Maybe we can implement a test in `shard_tries.rs` - I recall that unit tests on such level have access to private methods and fields.

Also I recommend to use `.iter().map(...).collect()` to convert trie changes to ops for code style reasons.

Reply:
Aha I understand, I will move it out of update_cache then and also add tests for this.

But then making `.test.map(...).collect()` won't make sense since I will place the creation of ops in an already existing loop.

Reply:
I've added a test for trie cache, it only tests insertion not deletion, since deletion does not happen immediately but it is placed in a queue.

Reply:
Hmm. Looks good already, but testing deletions is quite critical for us, because correct deletions allow us to make effective cache capacity 2x larger if I remember correctly.

Is it fine if I ask you to implement overriding `deletions_queue_capacity` in `TrieConfig`?
I think we can do it as follows:
* add new field `deletions_queue_capacity` to `TrieConfig` with default value `DEFAULT_SHARD_CACHE_DELETIONS_QUEUE_CAPACITY` (annotations allow us to use them)
* Instead of `ShardTries::test` call `ShardTries::new` with `TrieConfig` where this value is overridden by 0
* now deletions in `ShardTries` should actually delete values.

This is quite hacky setup but I really want to make sure that trie cache works correctly. It's even possible that deletions queue is not fully tested currently :) We can improve this code later.

Reply:
Yes that makes sense

Reply:
I will also add a test for testing if we cache value above the maximum value, since I think it is useful